rust/arrow/src/compute/kernels/boolean.rs

```diff
@@ -457,4 +517,42 @@ mod tests {
         assert_eq!(true, res.value(2));
         assert_eq!(false, res.value(3));
     }
+
+    fn assert_array_eq<T: ArrowNumericType>(
+        expected: PrimitiveArray<T>,
+        actual: ArrayRef,
+    ) {
+        let actual = actual
+            .as_any()
+            .downcast_ref::<PrimitiveArray<T>>()
+            .expect("Actual array should unwrap to type of expected array");
+
+        for i in 0..expected.len() {
+            if expected.is_null(i) {
+                assert!(actual.is_null(i));
+            } else {
+                assert_eq!(expected.value(i), actual.value(i));
+            }
+        }
+    }
+
+    #[test]
+    fn test_nullif_int_array() {
+        let a = Int32Array::from(vec![Some(15), None, Some(8), Some(1), Some(9)]);
+        let comp =
+            BooleanArray::from(vec![Some(false), None, Some(true), Some(false), None]);
+        let res = nullif(&a, &comp).unwrap();
+
+        let expected = Int32Array::from(vec![
+            Some(15),
+            None,
+            None, // comp true, slot 2 turned into null
+            Some(1),
+            // Even though comp array / right is null, should still pass through original value
+            // comp true, slot 2 turned into null
+            Some(9),
+        ]);
+
+        assert_array_eq::<Int32Type>(expected, Arc::new(res));
```

Comment:
Thanks for your changes, @velvia 

If `assert_eq!` is complaining, then we need to investigate why, because that is our canonical way of proving that two arrays are equal.

I think that the issue that `assert_eq` is flagging is that the function `nullif` is returning an arrow array that is out of spec: the `null_count` is equal to the number of elements (due to `Some(left.len()))`), but the actual null count of that array is 2. The array equality takes into account that number (something that `assert_array_eq` does not).

For reference, the way I use to check for these is to debug print the data:

```rust
println!("{:?}", expected.data());
println!("{:?}", result.data());
```

Reply:
How is the null_count derived?  Shouldn't it look at the nullity bitmap?  If it is actually checking that the data is zero, then we'd need to modify the array data as part of the nullif() implementation, which would be more expensive and possibly hard to do in a generic way, no?

Reply:
Ah, never mind, I see what you mean that we are passing in `Some(left.len())`....   will see about how to derive the null count from the new bitmap.

Reply:
The null_count is fixed... so the test can use `assert_eq!`...  now onto the offsets, which is slightly trickier